packages/rest/tests/proof.test.ts

```diff
@@ -238,10 +240,47 @@ describe('ProofController', () => {
     })
   })
 
+  describe('Proof WebSocket event', () => {
+    test('should return proof event sent from test agent to websocket client', async () => {
+      expect.assertions(1)
+
+      const proofRecord = new ProofRecord({
+        id: 'testest',
+        state: ProofState.ProposalSent,
+        threadId: 'random',
+      })
+
+      // Start client and wait for it to be opened
+      const client = new WebSocket('ws://localhost:3000')
+      await new Promise((resolve) => client.once('open', resolve))
+
+      // Start promise to listen for message
+      const waitForEvent = new Promise((resolve) =>
+        client.on('message', (data) => {
+          client.terminate()
+          resolve(JSON.parse(data as string))
+        })
+      )
+
+      bobAgent.events.emit<ProofStateChangedEvent>({
+        type: ProofEventTypes.ProofStateChanged,
+        payload: {
+          previousState: null,
+          proofRecord,
+        },
+      })
+
+      // Wait for event on WebSocket
+      const event = await waitForEvent
+      expect(event).toHaveProperty('type', 'ProofStateChanged')
```

Comment:
Maybe we can add a test for the structure so we know the json was serialized correctly and all fields are present?

Reply:
On it